modules/broker-auth/src/main/java/org/wso2/broker/auth/user/impl/UserStoreManagerImpl.java

```diff
@@ -0,0 +1,118 @@
+ */
+public class UserStoreManagerImpl implements UserStoreManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            org.wso2.broker.auth.user.impl.UserStoreManagerImpl.class);
+
+    /**
+     * Store the map of userRegistry
+     */
+    private static Map<String, User> userRegistry = new ConcurrentHashMap<>();
+
+    public UserStoreManagerImpl() throws ConfigurationException {
+        Path usersYamlFile;
+        String usersFilePath = System.getProperty(BrokerAuthConstants.SYSTEM_PARAM_USERS_CONFIG);
+        if (usersFilePath == null || usersFilePath.trim().isEmpty()) {
+            // use current path.
+            usersYamlFile = Paths.get("", BrokerAuthConstants.USERS_FILE_NAME).toAbsolutePath();
+        } else {
+            usersYamlFile = Paths.get(usersFilePath).toAbsolutePath();
+        }
+        ConfigProvider configProvider = ConfigProviderFactory.getConfigProvider(usersYamlFile, null);
+        UsersFile usersFile = configProvider
+                .getConfigurationObject(BrokerAuthConstants.USERS_CONFIG_NAMESPACE, UsersFile.class);
+        if (usersFile != null) {
+            List<UserConfig> usersList = usersFile.getUserConfigs();
+            for (UserConfig userConfig : usersList) {
+                if (userConfig != null && userConfig.getUsername() != null) {
+                    User user = new User(userConfig.getUsername(), userConfig.getPassword().toCharArray(), userConfig
+                            .getRoles());
+                    userRegistry.put(userConfig.getUsername(), user);
+                } else {
+                    LOGGER.error("User or username can not be null");
+                }
+            }
+        }
+    }
+
+    /**
+     * Authenticate given user with credentials.
+     *
+     * @param userName    userName
+     * @param credentials Credentials
+     * @return Authentication result
+     * @throws BrokerAuthException Exception throws when authentication failed.
+     */
+    @Override
+    public boolean authenticate(String userName, char... credentials) throws BrokerAuthException {
+        User user = userRegistry.get(userName);
+        if (user == null) {
+            throw new BrokerAuthException("User not found for the given username.");
+        } else {
+            if (credentials != null && Arrays.equals(credentials, user.getPassword())) {
+                return true;
+            } else {
+                throw new BrokerAuthException("Password did not match with the configured user");
+            }
+        }
+    }
+
+    /**
+     * Retrieve the list of userRegistry for given username.
+     *
+     * @param userName user name
+     * @return List of roles
+     */
+    public Set<String> getUserRoleList(String userName) {
+        User user = userRegistry.get(userName);
+        if (user != null) {
+            return Collections.unmodifiableSet(new HashSet<>(user.getRoles()));
```

Comment:
Can there be duplicate roles in user.getRoles? If not can we use a set in User class itself for roles? If so we can create an unmodifiable view for the roles in the User class itself in the constructor. Then we can returns the unmodifiable view when we invoke User#getRoles